float64 stamp
float32[8] dis_arr
